### ISSUE TITLE
internal/i3 label-focused-foreground stopped working

### ISSUE THREAD

**Issue by ecly** [jaagr/polybar#937]
I sincerely hope that this is not me being an idiot, but without touching my config and merely having updated polybar potentially several times, I've now noticed that `label-focused-foreground` no longer affects the foreground of the focused workspace for module `internal/i3`. Surprisingly however, `label-focused-underline` works just fine.  The entire relevant section of my config can be seen [here](https://github.com/ecly/dotfiles/blob/master/.config/polybar/config#L111-#L172). Is this the case for anyone else with most recent polybar? (I use polybar-git from AUR)

**Comment by patrick96** [comment]
I don't think we changed anything in the i3 module. I am also not able to reproduce this on my end, can you post a screenshot of the bar where this happens. Also which version are you running?

**Comment by ecly** [comment]
I'm even starting to doubt myself that it was ever working. But it may not be related to the i3 module but rather partial coloring of strings. I use partial coloring of icons like this:

```
ws-icon-0 = "1;%{F#f00}%{F-} web"
```

I believe the text that wasn't enclosed in {F} tags used to get nicely affected by foreground color declarations, but that doesn't seem to happen for me anymore. Has there been any changes regarding that perhaps?

**Comment by ecly** [comment]
Oh, and version info:

```bash
$ polybar --version
polybar 3.1.0-4-gd3abcc9

Features: +alsa +curl +i3 +mpd +network
```

**Comment by patrick96** [comment]
Yeah, I know what this is. 
It is not that you use partial color codes in the tag but the color formatting tag itself is the problem. Polybar generates the following code:
```
%{F#5F819D} %{F#f00}%{F-} web %{F-}
```
The problem with this is, after the first reset tag (`%{F-}`) directly after the icon, the foreground color is reset to the foreground color of the bar and not the foreground color of the label (because that's what the reset tag does).
For a short period of time the `polybar-git` package had code in it where this actually worked, but it broke a lot of other things and we reverted the changes and tried to fix things in a non breaking way. Seems like we missed this kind of bug.

@NBonaparte, @jaagr Looks like the color stacks were a good idea nonetheless. Apart from the stack approach, I don't see any other way to fix this without letting the parser know about what labels and formats there are and what colors they have set. But since the stacks broke a lot of things and introduced the expectation that tags and their reset variants should appear in pairs, I think we should make a fix that uses stacks part of the next major release.

**Comment by NBonaparte** [comment]
OK. In #639 I recommended adding another tag for popping single items off the stack; maybe we could implement that?

**Comment by patrick96** [comment]
Yeah that's definitely what I thought of as well. I can't think of a way to implement this in a non-breaking way though. And if we have to break existing functionality anyways, we should take the opportunity and define some proper syntax rules (like tags only in pairs), this way something like this won't come back to bite us in the ass later.
